### DISCUSSION TITLE
Odroid M1 (rockchip RKNN) instead of coral

### DISCUSSION THREAD

**Discussion by coendaar** [blakeblackshear/frigate#3304] [Ask A Question] [answered]
With  coral sold out everwhere, the odroid M1 board with integrated RKNN (rockchip neural network 0.8TOPS) seems like a great lowcost alternatieve especially with the M2-80 NVME slot. 
Will this work?

**Comment by NickM-27** [comment, marked as answer]
0.8 TOPS seems a little low compared to the [corals 4](https://coral.ai/docs/edgetpu/benchmarks/) but hopefully supporting that chip could support others as well. 

I believe that the main point of concern is how the rockchip neural network interfaces with tensorflow lite models, it is much easier for it to interface with the models directly as opposed to running through an SDK which makes supporting it more complicated and difficult.

**Comment by RLutsch** [comment]
this board seems similar to the rockpi 3a, I couldn't get it to use the armnn library to work with tensorflow lite, also the rock pi 4c should run 2.4 tops but same story, can't really figure out all the pieces it needs to work. be great to have some pointers there. also soon the rock 5a will launch and can handle 6 tops but pretty sure it will have the same support issues

**Reply by StuartIanNaylor** [reply]
Try this but just change with what is correct for sudo bash -c 'echo "libmali-valhall-g610-g6p0-x11.so" > /etc/OpenCL/vendors/mali.icd'
I am using a rk3588 and it works but the Arm examples seem very sensitive to distro.
The wav2letter example is working on both CpuAcc & GpuAcc still waiting for the 6Tops NPU to get plumbed in as still msg's in dmesg

https://github.com/StuartIanNaylor/rock5b-wav2letter-bench

Getting about 50x realtime with the CPU & 45x realtime with the GPU if preprocessing the MFCC on the RK3588 I think they approximate to around 2 tops each
